### PR TITLE
Close sessions on Keeper shutdown

### LINKED ISSUE #41076: Remove all connected sessions during keeper shutdown
Sessions in clickhouse-keeper are local entity for each quorum participant but some information about session are global for the state of keeper: session_ids and corresponding ephemeral nodes. It make sense to send close requests to RAFT during server shutdown so all connected sessions will be closed properly and corresponding ephemeral nodes will be removed from state.

### PR DESCRIPTION
**Pull request by antonio2368**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
keeper-improvement: properly close active sessions during shutdown

Close #41076 

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Coordination/KeeperDispatcher.cpp [hunk 1]
```diff
@@ -368,10 +365,52 @@ void KeeperDispatcher::shutdown()
...
+
+            session_to_response_callback.clear();
+        }
+
+        // if there is no leader, there is no reason to do CLOSE because it's a write request
+        if (hasLeader() && !close_requests.empty())
+        {
+            LOG_INFO(log, "Trying to close {} session(s)", close_requests.size());
+            const auto raft_result = server->putRequestBatch(close_requests);
+            auto sessions_closing_done = std::make_shared<Poco::Event>();
```
Comment (alesapin) [inline src/Coordination/KeeperDispatcher.cpp:398]: Oh, future/promises are much better...

File: src/Coordination/KeeperDispatcher.cpp [hunk 2]
```diff
@@ -368,10 +365,52 @@ void KeeperDispatcher::shutdown()
...
+            }
+
+            session_to_response_callback.clear();
+        }
+
+        // if there is no leader, there is no reason to do CLOSE because it's a write request
+        if (hasLeader() && !close_requests.empty())
+        {
+            LOG_INFO(log, "Trying to close {} session(s)", close_requests.size());
+            const auto raft_result = server->putRequestBatch(close_requests);
```
Comment (alesapin) [inline src/Coordination/KeeperDispatcher.cpp:397]: batching close requests! Smart!

**Review by alesapin — APPROVED** [review]
LGTM

File: src/Coordination/KeeperDispatcher.cpp (hunk 1 above)
Comment (antonio2368) [reply src/Coordination/KeeperDispatcher.cpp]: agree, changed

**Pull request merged**